pkg/scheduler/framework/runtime/framework.go

```diff
@@ -711,18 +711,22 @@ func (f *frameworkImpl) RunPostFilterPlugins(ctx context.Context, state *framewo
 	}()
 
 	statuses := make(framework.PluginToStatus)
+	// `result` records the last meaningful(non-noop) PostFilterResult.
+	var result *framework.PostFilterResult
 	for _, pl := range f.postFilterPlugins {
 		r, s := f.runPostFilterPlugin(ctx, pl, state, pod, filteredNodeStatusMap)
 		if s.IsSuccess() {
 			return r, s
 		} else if !s.IsUnschedulable() {
 			// Any status other than Success or Unschedulable is Error.
 			return nil, framework.AsStatus(s.AsError())
+		} else if r != nil && r.Mode() != framework.ModeNoop {
```

Comment:
perhaps `r.Mode` could support `r==nil`

Reply:
Nope, r.Mode() is a simplified form of r.NominatingInfo.Mode(). Only NominatingInfo has the logic for nil tolerance. `r`, which is PostFilterResult, doesn't have.

Reply:
got it.